content/pl/codebase.md

```diff
@@ -0,0 +1,17 @@
+## I. Codebase
+### Jedno źródło kodu śledzone systemem kontroli wersji, wiele wdrożeń
+
+Aplikacja napisana wg dwunastu aspektów jest zawsze śledzona w systemie kontroli wersji takim jak [Git](http://git-scm.com/), [Mercurial](http://mercurial.selenic.com/), czy [Subversion](http://subversion.apache.org/). Miejsce, w którym trzymany i rewizjonowany jest kod nazywane jest *repozytorium kodu źródłowego*, często skracane do samego *code repo*, albo po prostu *repo*.
+
+*Codebase* jest więc niczym innym jak pojedynczym repo (w przypadku zcentralizowanego systemu kontroli wersji jak Subversion), albo zestawem repozytoriów, które współdzielą tzw. root commit. (w przypadku zdecentralizowanego systemu jak Git).
+
+![One codebase maps to many deploys](/images/codebase-deploys.png)
+
+Aplikacja powinna zawsze odzwierciedlać codebase:
+
+* Jeśli jest wiele źródeł, z których pobierany jest kod, nie jest to aplikacja, a system rozproszony. Każdy komponent w systemie rozproszonym jest aplikacją i każdy możte z osobna spełniać wszystkie 12 aspektów.
+* Wiele aplikacji dzielących ten sam kod jest naruszeniem 12 aspektów. Roziwązaniem tej sytuacji może być wyizolowanie współdzielonego kodu do bibliotek, które mogą być załączane przez tzw. [dependency manager](./dependencies).
+
+Jest tylko jeden codebase na aplikację, jednak może być wiele różnych wdrożeń tej samej aplikacji.  *Deploy* (z ang. wdrożenie) jest działającą instancją aplikacji. Zazwyczaj jest to wersja produkcyjna i jedna lub więcej w środowisku przedprodukcyjnym. Dodatkowo każdy developer posiada kopię aplikacji działającą na swoim lokalnym środowisku developerskim, która również kwalifikuje się jako deploy.
```

Comment:
1. Znowu wstawka tego angielskiego _codebase_. Musisz wybrać jakaś koncepcję, bo na razie jest niespójnie. Albo wszędzie używać angielskiego i tylko na początku wyjaśnić co to jest, ale wtedy też zaznaczaj angielskie wstawki np. kursywą. 

2. _Jednak może być wiele wdrożen_ -> _możliwe jest jest wiele wdrożen tej samej aplikacji_

3.  _Zazwyczaj jest to wersja produkcyjna i jedna lub więcej w środowisku przedprodukcyjnym_ -> źle się to czyta. Może "Aplikacja zazwyczaj posiada wersję produkcyjną oraz jedną lub więcej przedprodukcyjnych"

4. _na swoim lokalnym środowisku_ -> _**w** swoim lokalnym środowisku_